### PR TITLE
v1.14 Backports 2024-03-12

### PR DESCRIPTION
**Pull request by jibi**
* [x] #30988 (@tommyp1ckles)
 * [x] #31014 (@tommyp1ckles)
 * [x] #31159 (@pchaigno)
 * [x] #31182 (@aanm)
 * [x] #31196 (@aanm)
 * [x] #31060 (@sayboras)
 * [x] #31179 (@YutaroHayakawa)
 * [x] #31218 (@YutaroHayakawa)
 * [x] #31015 (@learnitall)

PRs skipped due to conflicts:

 * #30686 (@cjvirtucio87)
 * #31292 (@chaunceyjiang)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 30988 31014 31159 31182 31196 31060 31179 31218 31015
```

### CODE REVIEW

**Comment by jibi** [comment]
/test-backport-1.14

File: CODEOWNERS [hunk 1]
```diff
@@ -217,6 +217,11 @@
 /.github/workflows/ @cilium/github-sec @cilium/ci-structure
 /.github/workflows/*datapath*.yaml @cilium/sig-datapath @cilium/github-sec @cilium/ci-structure
 /.github/workflows/*perf*.yaml @cilium/sig-scalability @cilium/github-sec @cilium/ci-structure
+/.gitignore @cilium/tophat
+/.golangci.yaml @cilium/ci-structure
+/.mailmap @cilium/tophat
+/.nvim @cilium/contributing
+/.vscode @cilium/contributing
```
Comment (aanm) [inline CODEOWNERS]: This is not part of my commit :eyes:

**Review by aanm — CHANGES_REQUESTED** [review]

File: CODEOWNERS (hunk 1 above)
Comment (jibi) [reply CODEOWNERS]: fixed

**Comment by sayboras** [comment]
>  * #30686 (@cjvirtucio87)
> * #31292 (@chaunceyjiang)

Noted and thanks, I will take care of these two backports for 1.14.

**Review by sayboras — APPROVED** [review]
Thanks and looks good for my commit.

**Comment by jibi** [comment]
/test-backport-1.14

**Review by aanm — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]
My PR looks good. Thanks!

**Review by YutaroHayakawa — APPROVED** [review]
Thanks!

**Review by learnitall — APPROVED** [review]
Thank you!

**Review by tommyp1ckles — APPROVED** [review]

**Review by dylandreimerink — APPROVED** [review]

**Pull request merged**